src/util/psbt/map/global.rs

```diff
@@ -113,6 +155,54 @@ impl Map for Global {
             });
         }
 
+        // BIP 174: The Combiner must remove any duplicate key-value pairs, in accordance with
+        //          the specification. It can pick arbitrarily when conflicts occur.
+
+        // Keeping the highest version
+        self.version = cmp::max(self.version, other.version);
+
+        // Merging xpubs
+        for (xpub, (fingerprint1, derivation1)) in other.xpub {
+            match self.xpub.entry(xpub) {
+                Entry::Vacant(entry) => {
+                    entry.insert((fingerprint1, derivation1));
+                },
+                Entry::Occupied(mut entry) => {
+                    // Here in case of the conflict we select the version with algorithm:
+                    // 1) if everything is equal we do nothing
+                    // 2) report an error if
+                    //    - derivation paths are equal and fingerprints are not
+                    //    - derivation paths are of the same length, but not equal
+                    //    - derivation paths has different length, but the shorter one
+                    //      is not the strict prefix of the longer one
+                    // 3) choose longest derivation otherwise
+
+                    let (fingerprint2, derivation2) = entry.get().clone();
+
+                    let len_diff_i64 = derivation1.len() as i64 - derivation2.len() as i64;
+                    let len_diff = len_diff.abs() as usize;
+                    if derivation1 == derivation2 && fingerprint1 == fingerprint2
+                    {
+                        continue
+                    }
+                    else if
+                        len_diff_i64 < 0 &&
+                        derivation1[..] == derivation2[derivation2.len() - len_diff..]
```

Comment:
Let's say `len1 = 2` and `len2 = 5`. That means `len_diff = 3`. This would make us compare `derivation1[0..2]` (of length 2) with `derivation2[2..5]` (of length 3). This doesn't seem to be  a valid suffix check as it would return false even though derivation 1 might be a suffix of derivation 2. 

I'd propose a more direct approach as I mentioned in another comment:

```rust
derivation1[..] == derivation2[(derivatio2.len()-derivation1.len())..]
```

I hope I didn't make any errors, I'm kinda tired but wanted to respond asap. So please double check.

Reply:
Yes, this seems right. Sorry, was sleeping already. Pushed the latest version

Reply:
Thx, looks good now :)